phac-nml/irida-next
# Fix transfer samples errors

## What does this PR do and why?
Moved sample transfer errors into a dialog, like in sample cloning.

## Screenshots or screen recordings
![Screenshot from 2024-03-01 10-08-07](https://github.com/phac-nml/irida-next/assets/325703/311b6ae9-e806-4cca-a969-e58ffb3081a8)
![Screenshot from 2024-03-01 10-46-45](https://github.com/phac-nml/irida-next/assets/325703/a959e97b-b840-4951-bf8e-82957abed4da)

## How to set up and validate locally
Test sample error:
1. Login to irida-next.
2. Create a new project. Call it `Project1`.
3. Create a new sample within `Project1`. Call it `Sample1`.
4. Create another new project. Call it `Project2`.
5. Create a new sample within `Project2`. Call it `Sample1`.
6. Select `Sample1` within the samples table and click the `Transfer samples` button.
7. The `Transfer Samples` dialog should appear. Select `Project1` from the drop-down and click the `Submit` button.
8. Verify the error explaining the sample name conflict is being displayed. 

Test validation error:
1. Login to irida-next.
2. Create a new project. Call it `Project1`.
3. Create a new sample within `Project1`. Call it `Sample1`.
4. Create another new project. Call it `Project2`.
5. Create a new sample within `Project2`. Call it `Sample1`.
6. Select `Sample1` within the samples table.
7. Clear your browser session storage.
8. Click the `Transfer samples` button.
9. The `Transfer Samples` dialog should appear. Select `Project1` from the drop-down and click the `Submit` button.
10. Verify the error explaining the no samples have been selected is being displayed. 

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
